editor/js/editor-libs/telemetry.js

```diff
@@ -0,0 +1,68 @@
+import { postParentMessage } from "./events.js";
+
+const ACTION_COUNTS_KEY = "action-counts";
+
+function readActionsCounts() {
+  try {
+    const value = JSON.parse(localStorage.getItem(ACTION_COUNTS_KEY));
+    if (value && value.href === window.location.href && value.counts) {
+      return value.counts;
+    }
+  } catch (e) {
+    window.console.warn("Unable to read action counts from localStorage", e);
+  }
+
+  return {};
+}
+
+function persistActionCounts(counts) {
+  try {
+    localStorage.setItem(
+      ACTION_COUNTS_KEY,
+      JSON.stringify({
+        href: window.location.href,
+        counts,
+      })
+    );
+  } catch (e) {
+    window.console.warn("Unable to write action counts to localStorage", e);
+  }
+}
+
+const actionCounts = readActionsCounts();
+let lastAction = null;
+
+/**
+ * @param {string} key
+ * @param {boolean} once
+ */
+export function recordAction(key, deduplicate = false) {
+  actionCounts[key] = actionCounts[key] ?? 0;
+
+  if (deduplicate && key === lastAction) {
+    return;
+  } else {
+    lastAction = key;
+  }
+
+  let source = `${key} -> ${actionCounts[key]}`;
+  actionCounts[key]++;
```

Comment:
Wouldn't it be better to push it 1 line higher? Currently, this function kind of reports the amount of previously performed actions, because the first message will be `focus -> 0` instead of `focus -> 1`.

Reply:
I intentionally used 0-based, but thinking about it again, 1-based probably makes more sense.

Reply:
Resolved via 27b660c9464b66c96de1515bb5c2dcbd93359ff5.